src/libs/leaderboard.js

```diff
@@ -0,0 +1,158 @@
+import _ from 'lodash';
+
+/**
+ * Initialize match counting for a user A.
+ * If result of A vs B has been counted, mark matchCounting[B] = true. (Else false)
```

Comment:
Incorrect documentation? Your matchCounting is an object containing an object according to your code below: `matchCounting[A][B]`